### PR TITLE
Add python extensions for new IPeak methods

### PR DESCRIPTION
**Pull request by samueljackson92**
#20441 added two new methods to `IPeak` but did not export them to python.

**To test:**
 - Load/Create a PeaksWorkspace and then access the peak via the python interface
 - The new methods to be checked are:
   - `getEnergy()`
   - `getIntensityOverSigma()` 

*There's no issue associated with this PR.*

**Release Notes** 
[See here](https://github.com/mantidproject/mantid/pull/21556/commits/aa1fa6c3f92cbcb00906d74cc26686489119ab20)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp [hunk 1]
```diff
@@ -128,6 +128,8 @@ void export_IPeak() {
            "Get the initial (incident) neutron energy")
       .def("getFinalEnergy", &IPeak::getFinalEnergy, arg("self"),
            "Get the final neutron energy")
+      .def("getEnergy", &IPeak::getEnergy, arg("self"),
```
Comment (OwenArnold) [inline Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp]: Did you consider referring to this as a delta in energy, or is there a good reason why this is just referred to as the energy?

**Review by OwenArnold — COMMENTED** [review]
Did you consider adding tests to cover the python exports?

File: Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp (hunk 1 above)
Comment (samueljackson92) [reply Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp]: I did consider that, but this name was consistent with what the column in the slice viewer was called where the value was originally being calculated. I'm not sure which is better. `Energy` is a bit vague.

**Comment by samueljackson92** [comment]
> Did you consider adding tests to cover the python exports?

It appears that there are no tests at all for `IPeak` python bindings.

File: Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp (hunk 1 above)
Comment (OwenArnold) [reply Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp]: I think that @AndreiSavici should provide some advice here. Deprecating parts of the python api is a difficult process, so better to get it right first time if we can.

Comment (AndreiSavici) [reply Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp]: Not sure anyone is using that. But yes, it should be `DeltaE` or `EnergyTransfer`. You might want to change the C++ code as well

Comment (samueljackson92) [reply Framework/PythonInterface/mantid/api/src/Exports/IPeak.cpp]: The name in the SliceViewer should also be changed then.

File: Framework/DataObjects/src/Peak.cpp [hunk 2]
```diff
@@ -714,7 +714,9 @@ double Peak::getFinalEnergy() const { return m_finalEnergy; }
 double Peak::getInitialEnergy() const { return m_initialEnergy; }
 
 /** Get the difference between the initial and final neutron energy */
```
Comment (OwenArnold) [inline Framework/DataObjects/src/Peak.cpp]: In what units? I think there should be a comment about this. Can do as a follow up change.

File: Framework/PythonInterface/test/python/mantid/geometry/IPeakTest.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        self.assertEqual(self._peak.getL1(), expected_l1)
+
+    def test_get_l2(self):
+        det_id = 101
+        expected_l2 = 0.2444125610784556
+        self._peak.setDetectorID(det_id)
+        self.assertEqual(self._peak.getL2(), expected_l2)
+
+
+if __name__ == '__main__':
```
Comment (OwenArnold) [inline Framework/PythonInterface/test/python/mantid/geometry/IPeakTest.py]: 👍

**Review by OwenArnold — APPROVED** [review]

File: Framework/DataObjects/src/Peak.cpp (hunk 2 above)
Comment (samueljackson92) [reply Framework/DataObjects/src/Peak.cpp]: Nah might as well do it here. Initial and final are in meV so should also be in meV.

**Pull request merged**